playbooks/add-repo.yml

```diff
@@ -0,0 +1,111 @@
+---
+- name: Create Ceph Apt Repo
+  hosts: repo_server[0]
+  vars:
+    target: "{{ dest | default('/opt/ceph-repo') }}"
+  tasks:
+    - name: Check for required variables
+      assert:
+        that: item is defined
+      loop_control:
+        label: "Assert {{ item }} is defined"
+      with_items:
+        - ceph_repo_version
+        - repo_server_interface
+
+    - name: Install dependencies
+      package:
+        name: "{{ item }}"
+        state: installed
+      with_items:
+        - dpkg-dev
+        - dpkg-sig
+        - apache2
+        - rng-tools
+
+    - name: Ensure target dir
+      file:
+        path: "{{ item }}"
+        state: directory
+      with_items:
+        - "{{ target }}/dists/{{ ansible_distribution_release }}/main/binary-amd64"
+        - "{{ target }}/pool/main/c/ceph"
+
+    - name: Get Ceph Packages
+      command: >
+        rsync -v download.ceph.com::ceph/debian-{{ ceph_stable_release }}/pool/main/c/ceph/
+          --recursive --times --links --hard-links
+          --include='*{{ ceph_repo_version }}{{ ansible_distribution_release }}*amd64.deb'
+          --include='*/' --exclude='*'
+          {{ target }}/pool/main/c/ceph
+      args:
+        creates: "{{ target }}/pool/main/c/ceph/ceph_{{ ceph_repo_version }}{{ ansible_distribution_release }}_amd64.deb"
+
+    - name: Force version refresh
+      command: >
+        rsync -v download.ceph.com::ceph/debian-{{ ceph_stable_release }}/pool/main/c/ceph/
+          --recursive --times --links --hard-links
+          --include='*{{ ceph_repo_version }}{{ ansible_distribution_release }}*amd64.deb'
+          --include='*/' --exclude='*'
+          {{ target }}/pool/main/c/ceph
+      when: force | default(false)
+
+    - name: Link target to apache directory
+      file:
+        state: link
+        src: "{{ target }}"
+        path: "/var/www/html/debian-{{ ceph_stable_release }}"
+
+    - name: GPG Creation Script
```

Comment:
GPG creation and related tasks required for a local apt repo?

Reply:
Yes.  The repo is being served behind a web server so it is accessible to all ceph nodes.  Ceph-ansible is responsible for configuring the repository source on each node and it is expecting signed packages.